### PR TITLE
add docs for `name` attribute of `details`

### PR DESCRIPTION
**Pull request by chadluo**
### Description

Add explanation of `names` attribute of `details` element.

### Motivation

This attribute is listed in the [Browser compatibility table below](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details#browser_compatibility) but doesn't come with explanations.

### Additional details

The feature related to the attribute is supported in Chrome and Safari [since 17.2](https://webkit.org/blog/14787/webkit-features-in-safari-17-2/#:~:text=Exclusive%20accordions) (called 'Exclusive accordions'), but not in Firefox yet.

### Related issues and pull requests

N/A

### CODE REVIEW

File: files/en-us/web/html/element/details/index.md [hunk 1]
```diff
@@ -33,6 +33,12 @@ This element includes the [global attributes](/en-US/docs/Web/HTML/Global_attrib
 
     > **Note:** You have to remove this attribute entirely to make the details hidden. `open="false"` makes the details visible because this attribute is Boolean.
 
+- `name`
+
+  - : This attribute connects multiple `<details>` elements, such that at any time at most one of them will be open. When one `<details>` element twists open and its content is shown, other `<details>` elements with the same `name` will be closed. If the only open `<details>` of the `name` is closed, other `<details>` will remain closed.
```
Comment (chrisdavidmills) [inline files/en-us/web/html/element/details/index.md:38]:
```suggestion
  - : This attribute gives the `<details>` element a _group name_. When multiple `<details>` elements are given the same name value, they all become part of the same group. Only one `<details>` element out of a group can be open at a time — when one `<details>` element twists open and its content is shown, an already open `<details>` element with the same `name` will be closed. If multiple `<details>` elements in a group are given the `open` attribute, only the first one in the source order will be rendered open.
```

File: files/en-us/web/html/element/details/index.md [hunk 2]
```diff
@@ -33,6 +33,12 @@ This element includes the [global attributes](/en-US/docs/Web/HTML/Global_attrib
...
+
+    > **Note:** Connected `<details>` elements don't have to be adjacent to each other.
```
Comment (chrisdavidmills) [inline files/en-us/web/html/element/details/index.md:40]:
```suggestion
    > **Note:** `<details>` elements don't have to be adjacent to one another in the source to be part of the same group.
```

**Review by chrisdavidmills — APPROVED** [review]
Hi @chadluo, and thank you so much for contributing to MDN. I think overall what you want to say makes sense. I've provided a couple of wording suggestions, to clarify meaning a bit.

What do you think?

**Review by chrisdavidmills — CHANGES_REQUESTED** [review]
Ooops, sorry, I was supposed to request changes...

File: files/en-us/web/html/element/details/index.md (hunk 1 above)
Comment (chadluo) [reply files/en-us/web/html/element/details/index.md:38]: Thanks @chrisdavidmills for reviewing and updating! And it's helpful to add the last part to describe the initialisation behaviour.

I'm considering whether it's a clearer style to describe the most important effect of the attribute first and then explain the details. For `name` here the most important part is it relates a few `<details>` together and interacting with one could change the state of others, which is what I want to highlight. WDYT like this:

```suggestion
  - : This attribute creates groups of `<details>` elements, and in one group only one `<details>` can be open at a time. The attribute gives a `<details>` element a _group name_, and `<details>` elements sharing the same name become part of the same group. When one `<details>` element twists open and its content is shown, an already open `<details>` element with the same `name` will be closed. If multiple `<details>` elements in a group are given the `open` attribute, only the first one in the source order will be rendered open.
```

Comment (chrisdavidmills) [reply files/en-us/web/html/element/details/index.md:38]: Hi @chadluo — stating the key use case to begin with, followed by the details, is a good way to do it, I'd like to think about making the copy a bit tighter. How about something like this:

- : This attribute enables multiple `<details>` elements to be connected, with only one open at a time. This allows developers to easily create UI features such as accordions without scripting.

  The `name` attribute specifies a _group name_ —  give multiple `<details>` elements the same name value to group them. Only one of the grouped `<details>` elements can be open at a time — opening one will cause another to close. If multiple grouped `<details>` elements are given the `open` attribute, only the first one in the source order will be rendered open.

?

Comment (chrisdavidmills) [reply files/en-us/web/html/element/details/index.md:38]: @chadluo cool, looks like you are happy with this. Feel free to make the changes, and I'll approve and merge.

**Review by chrisdavidmills — APPROVED** [review]
Perfect, thanks so much @chadluo!

I just added a small commit to codify some of the attribute names; I'll merge this now.

**Review by Pjrich1313 — COMMENTED** [review]
No comment

**Pull request merged**